pandas/core/algorithms.py

```diff
@@ -708,6 +708,10 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
 
     if sort:
         result = result.sort_values(ascending=ascending)
+    elif bins is None:
+        uniq = unique(values)
```

Comment:
so do
``uniq = uniques(values)`` for the non-EA case (above)
and do ``uniq = Series(values)._values.unique()`` for the EA case. though this means computing it twice. maybe have to work on that.

Reply:
I added it above, although it is not computed twice now. Did you mean it like this?